### ISSUE TITLE
"Undefined constant or missing '('" when parsing filename option for signature_cuda filter

### ISSUE THREAD

**Issue by yondonfu** [livepeer/FFmpeg#19]
```
[cudasign @ 00000171227a0ac0] [Eval @ 00000053e27fe960] Undefined constant or missing '(' in 'Usersrcurr.lpDataoffchain/out_ef9d7ece887e9e2ac576.tempfile.bin'
[cudasign @ 00000171227a0ac0] Unable to parse option value "Usersrcurr.lpDataoffchain/out_ef9d7ece887e9e2ac576.tempfile.bin"
[cudasign @ 00000171227a0ac0] [Eval @ 00000053e27fe960] Undefined constant or missing '(' in 'Usersrcurr.lpDataoffchain/out_ef9d7ece887e9e2ac576.tempfile.bin'
[cudasign @ 00000171227a0ac0] Unable to parse option value "Usersrcurr.lpDataoffchain/out_ef9d7ece887e9e2ac576.tempfile.bin"
[cudasign @ 00000171227a0ac0] Error setting option format to value Usersrcurr.lpDataoffchain/out_ef9d7ece887e9e2ac576.tempfile.bin.
[Parsed_signature_cuda_0 @ 00000171227a74c0] Error applying options to the filter.
Error initializing filter 'signature_cuda' with args 'filename=C:Usersrcurr.lpDataoffchain/out_ef9d7ece887e9e2ac576.tempfile.bin'
ERROR: filter.c:270] Unable to parse signature filters desc : Invalid argument
ERROR: encoder.c:269] Unable to open signature filter : Invalid argument
```

This error occurred on a few orchestrators, but not on all orchestrators. go-livepeer also proceeded to crash after this error was encountered.

Find below the information of a new machines that this occurred on:

https://discord.com/channels/423160867534929930/900820305130438716/900827779334012949

https://discord.com/channels/423160867534929930/900820305130438716/900829287207297034

https://discord.com/channels/423160867534929930/900820305130438716/900829925777494156

https://discord.com/channels/423160867534929930/900820305130438716/900830773232402442

Suspicious points:

- Weird filenames with certain slashes stripped?
- Could the file for the filename not exist? Does a non-existent file cause this issue?
- The error log contains `[Eval @ 00000053e27fe960]` which implies the use of the ffmpeg eval parser? Why is this being run in the first place? Could there be some sort of corruption causing the wrong code to be executed?

**Comment by yondonfu** [comment]
From @jailuthra in Discord:

> Ok I figured it out, it's a windows-only error, at least the string parsing one.

> All the logs I've seen with that particular error have the FFmpeg option parser complaining about a string like: Users<username>.lpDatamainnet/out_<id>.tempfile.bin
And those are all windows paths C:/Users/... 

> The problem is FFmpeg uses : as a delimiter between AVOptions, like here https://github.com/livepeer/FFmpeg/pull/14/files#diff-cb078e68911f0c4b339b9340fca008e3c3d131ac8f0ca62ba0d4950f72cb14aaR18803
And somehow seeing the delimiter in an unquoted string makes it go haywire, and run some string-to-math eval function 

> We didn't quote the filename before passing it to the AVOption :facepalm: 
> I think just adding quotes here should work https://github.com/livepeer/lpms/blob/master/ffmpeg/ffmpeg.go#L459

**Comment by iameli** [comment]
This does not seem to fix the problem, presuming I handled my build properly on https://github.com/livepeer/go-livepeer/pull/2063. Still getting:

```
[cudasign @ 000002b77f269280] [Eval @ 000000c8951fe520] Undefined constant or missing '(' in 'Usersiameli.lpDataoffchain/out_8d791743f660752bd23c.tempfile.bin'
[cudasign @ 000002b77f269280] Unable to parse option value "Usersiameli.lpDataoffchain/out_8d791743f660752bd23c.tempfile.bin"
[cudasign @ 000002b77f269280] [Eval @ 000000c8951fe520] Undefined constant or missing '(' in 'Usersiameli.lpDataoffchain/out_8d791743f660752bd23c.tempfile.bin'
[cudasign @ 000002b77f269280] Unable to parse option value "Usersiameli.lpDataoffchain/out_8d791743f660752bd23c.tempfile.bin"
[cudasign @ 000002b77f269280] Error setting option format to value Usersiameli.lpDataoffchain/out_8d791743f660752bd23c.tempfile.bin.
[Parsed_signature_cuda_0 @ 000002b77f25db80] Error applying options to the filter.
Error initializing filter 'signature_cuda' with args 'filename=C:\Users\iameli\.lpData\offchain/out_8d791743f660752bd23c.tempfile.bin'
ERROR: filter.c:270] Unable to parse signature filters desc : Invalid argument
ERROR: encoder.c:269] Unable to open signature filter : Invalid argument
ERROR: transcoder.c:186] Unable to open output : Invalid argument
Exception 0xc0000005 0x0 0x48 0x7ff64feb0f36
PC=0x7ff64feb0f36
```

I'm thinking it might be the backslashes in the file path rather than anything else? That'd explain why it's printing out as `Usersiameli.lpDataoffchain/out_8d791743f660752bd23c.tempfile.bin`. Most APIs in this context support translating forward slashes to backslashes as appropriate in Windows, perhaps we should try that?

**Comment by jailuthra** [comment]
Good catch! Looks like both `:` and `\` are special separators for AVOption parsing https://ffmpeg.org/ffmpeg-utils.html#Quoting-and-escaping.

So we need to set file paths in the options like `signature=filename='C\:\\Users\\iameli\\.lpData\\..'` according to https://stackoverflow.com/a/14526635

Manually setting that path on my machine worked out. But for the proper solution, we'll have to write a custom function to escape golang-generated paths for `:` and `\`, so that FFmpeg can understand it.

**Comment by yondonfu** [comment]
Closed by livepeer/lpms#268
